File: backend/src/main/java/com/vehicle/maintenance/dto/maintenance_plan/MaintenancePlanActivityResponse.java
package com.vehicle.maintenance.dto.maintenance_plan;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MaintenancePlanActivityResponse {

    private Long id;

    /** Si la línea proviene de copiar la plantilla: id de {@code maintenance_plan_activities}. */
    private Long clonedFromPlanActivityId;

    private String nombre;
    private String tipo;
    private Integer intervaloKm;
    private Integer intervaloMeses;
    private Boolean isActive;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: app_mantenimiento/lib/models/maintenance_plan/maintenance_plan_activity_response.dart
class MaintenancePlanActivityResponse {
  final int id;
  final int? clonedFromPlanActivityId;
  final String nombre;
  final String tipo;
  final int? intervaloKm;
  final int? intervaloMeses;
  final bool? isActive;

  MaintenancePlanActivityResponse({
    required this.id,
    this.clonedFromPlanActivityId,
    required this.nombre,
    required this.tipo,
    this.intervaloKm,
    this.intervaloMeses,
    this.isActive,
  });

  factory MaintenancePlanActivityResponse.fromJson(
          Map<String, dynamic> json) =>
      MaintenancePlanActivityResponse(
        id: (json['id'] as num?)?.toInt() ?? 0,
        clonedFromPlanActivityId: (json['clonedFromPlanActivityId'] as num?)
            ?.toInt(),
        nombre: json['nombre']?.toString() ?? '',
        tipo: json['tipo']?.toString() ?? '',
        intervaloKm: (json['intervaloKm'] as num?)?.toInt(),
        intervaloMeses: (json['intervaloMeses'] as num?)?.toInt(),
        isActive: json['isActive'] as bool?,
      );
}
